examples/coverage/default.nix

```
@@ -0,0 +1,38 @@
+{
+  stdenv,
+  lib,
+  hax,
+  coqPackages,
+  gnused,
+  craneLib,
+  bat,
+  coqGeneratedCore ? import ../../proof-libs/coq/coq {inherit stdenv coqPackages;},
+}:
+let
+  commonArgs = import ../commonArgs.nix {inherit craneLib lib;};
+  cargoArtifacts = craneLib.buildDepsOnly commonArgs;
+in
+  craneLib.mkCargoDerivation (commonArgs // {
+      inherit cargoArtifacts;
+      pname = "coverage";
+      doCheck = false;
+      buildPhaseCargoCommand = ''
+        cd examples/coverage/
+        cargo hax into coq
+	cd proofs/coq/extraction/
+	echo -e "-R ${coqGeneratedCore}/lib/coq/user-contrib/Core Core\n$(cat _CoqProject)" > _CoqProject
+	coq_makefile -f _CoqProject -o Makefile
+	make
+	cd ../../../../../
```

Running `cargo hax into coq` in the `extraction` will work fine here

Also, you could just wrap in a sub shell so that you don't need to bother with wierd `cd ..`s

```suggestion
        (
          cd examples/coverage/proofs/coq/extraction/
          cargo hax into coq
	  echo -e "-R ${coqGeneratedCore}/lib/coq/user-contrib/Core Core\n$(cat _CoqProject)" > _CoqProject
	  coq_makefile -f _CoqProject -o Makefile
	  make
	)
```